xonsh/tools.py

```diff
@@ -763,20 +763,6 @@ def is_int(x):
     return isinstance(x, int)
 
 
-def is_int_as_str(x):
-    """
-    Tests if something is an integer
-    If not a string to begin with is automatically false
-    """
-    if isinstance(x, str):
-        try:
-            int(x)
-        except ValueError:
-            return False
-        return True
-    return False
```

Comment:
I don't see a reason to remove these functions or their tests, even if they are not used.  They could still be useful to users or us in the future for other environment variable validators.